test/features/features/composer/recipient_composer_widget_test.dart

```diff
@@ -0,0 +1,184 @@
+      return GetMaterialApp(
+        localizationsDelegates: const [
+          AppLocalizationsDelegate(),
+          GlobalMaterialLocalizations.delegate,
+          GlobalWidgetsLocalizations.delegate,
+          GlobalCupertinoLocalizations.delegate,
+        ],
+        supportedLocales: LocalizationService.supportedLocales,
+        home: Scaffold(body: child),
+      );
+    }
+
+    testWidgets('RecipientComposerWidget renders correctly', (tester) async {
+      final listEmailAddress = <EmailAddress>[];
+
+      final widget = makeTestableWidget(
+        child: RecipientComposerWidget(
+          prefix: prefix,
+          listEmailAddress: listEmailAddress,
+          imagePaths: imagePaths,
+          maxWidth: 360,
+          keyTagEditor: keyEmailTagEditor,
+        ),
+      );
+
+      await tester.pumpWidget(widget);
+
+      await tester.pumpAndSettle();
+
+      final recipientComposerWidgetFinder = find.byType(RecipientComposerWidget);
+
+      expect(recipientComposerWidgetFinder, findsOneWidget);
+    });
+
+    testWidgets('RecipientComposerWidget renders list email address correctly', (tester) async {
+      final listEmailAddress = <EmailAddress>[
+        EmailAddress(null, 'test1@example.com'),
+        EmailAddress(null, 'test2@example.com'),
+      ];
+
+      final widget = makeTestableWidget(
+        child: RecipientComposerWidget(
+          prefix: prefix,
+          listEmailAddress: listEmailAddress,
+          imagePaths: imagePaths,
+          maxWidth: 360,
+          keyTagEditor: keyEmailTagEditor,
+        ),
+      );
+
+      await tester.pumpWidget(widget);
+
+      await tester.pumpAndSettle();
+
+      expect(find.byType(RecipientTagItemWidget), findsNWidgets(2));
+      expect(find.text('test1@example.com'), findsOneWidget);
+      expect(find.text('test2@example.com'), findsOneWidget);
+    });
+
+    testWidgets('RecipientTagItemWidget should have a `maxWidth` equal to the RecipientComposerWidget\'s `maxWidth`', (tester) async {
+      final listEmailAddress = <EmailAddress>[
+        EmailAddress('test1', 'test1@example.com'),
+      ];
+
+      final widget = makeTestableWidget(
+        child: RecipientComposerWidget(
+          prefix: prefix,
+          listEmailAddress: listEmailAddress,
+          imagePaths: imagePaths,
+          maxWidth: 360,
+          keyTagEditor: keyEmailTagEditor,
+        ),
+      );
+
+      await tester.pumpWidget(widget);
+
+      await tester.pumpAndSettle();
+
+      final recipientTagItemWidgetFinder = find.byKey(Key('recipient_tag_item_${prefix.name}_0'));
+
+      final Container recipientTagItemWidget = tester.widget(recipientTagItemWidgetFinder);
+
+      expect(recipientTagItemWidgetFinder, findsOneWidget);
+      expect(recipientTagItemWidget.constraints!.maxWidth, 360);
+    });
+
+    testWidgets('RecipientTagItemWidget should have all the components (AvatarIcon, Label, DeleteIcon)', (tester) async {
```

Comment:
This case failed, but in Flutter Inspector still see components well
```dart
    testWidgets('when email address is too long \n'
        'RecipientTagItemWidget should have all the components (AvatarIcon, Label, DeleteIcon)', (tester) async {
      final listEmailAddress = <EmailAddress>[
        EmailAddress(null, 'test123456789123456789@example.com'),
      ];

      final widget = makeTestableWidget(
        child: RecipientComposerWidget(
          prefix: prefix,
          listEmailAddress: listEmailAddress,
          imagePaths: imagePaths,
          maxWidth: 392.7,
          keyTagEditor: keyEmailTagEditor,
        ),
      );

      await tester.pumpWidget(widget);

      await tester.pumpAndSettle();

      final recipientTagItemWidgetFinder = find.byKey(Key('recipient_tag_item_${prefix.name}_0'));
      final labelRecipientTagItemWidgetFinder = find.byKey(Key('label_recipient_tag_item_${prefix.name}_0'));
      final deleteIconRecipientTagItemWidgetFinder = find.byKey(Key('delete_icon_recipient_tag_item_${prefix.name}_0'));
      final avatarIconRecipientTagItemWidgetFinder = find.byKey(Key('avatar_icon_recipient_tag_item_${prefix.name}_0'));

      final Size recipientTagItemWidgetSize = tester.getSize(recipientTagItemWidgetFinder);
      final Size labelRecipientTagItemWidgetSize = tester.getSize(labelRecipientTagItemWidgetFinder);
      final Size deleteIconRecipientTagItemWidgetSize = tester.getSize(deleteIconRecipientTagItemWidgetFinder);
      final Size avatarIconRecipientTagItemWidgetSize = tester.getSize(avatarIconRecipientTagItemWidgetFinder);

      log('recipient_composer_widget_test::main: TagSize = $recipientTagItemWidgetSize | LabelTagSize = $labelRecipientTagItemWidgetSize | DeleteIconTagSize = $deleteIconRecipientTagItemWidgetSize | AvatarIconTagSize = $avatarIconRecipientTagItemWidgetSize');

      expect(labelRecipientTagItemWidgetFinder, findsOneWidget);
      expect(deleteIconRecipientTagItemWidgetFinder, findsOneWidget);
      expect(avatarIconRecipientTagItemWidgetFinder, findsOneWidget);

      expect(
          labelRecipientTagItemWidgetSize.width + deleteIconRecipientTagItemWidgetSize.width + avatarIconRecipientTagItemWidgetSize.width,
          lessThanOrEqualTo(recipientTagItemWidgetSize.width)
      );
    });
```

Reply:
error:
```
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following TestFailure was thrown running a test:
Expected: exactly one matching candidate
  Actual: _KeyWidgetFinder:<Found 0 widgets with key [<'avatar_icon_recipient_tag_item_to_0'>]: []>
   Which: means none were found but one was expected

```

Reply:
> This case failed, but in Flutter Inspector still see components well
> 
> ```dart
>     testWidgets('when email address is too long \n'
>         'RecipientTagItemWidget should have all the components (AvatarIcon, Label, DeleteIcon)', (tester) async {
>       final listEmailAddress = <EmailAddress>[
>         EmailAddress(null, 'test123456789123456789@example.com'),
>       ];
> 
>       final widget = makeTestableWidget(
>         child: RecipientComposerWidget(
>           prefix: prefix,
>           listEmailAddress: listEmailAddress,
>           imagePaths: imagePaths,
>           maxWidth: 392.7,
>           keyTagEditor: keyEmailTagEditor,
>         ),
>       );
> 
>       await tester.pumpWidget(widget);
> 
>       await tester.pumpAndSettle();
> 
>       final recipientTagItemWidgetFinder = find.byKey(Key('recipient_tag_item_${prefix.name}_0'));
>       final labelRecipientTagItemWidgetFinder = find.byKey(Key('label_recipient_tag_item_${prefix.name}_0'));
>       final deleteIconRecipientTagItemWidgetFinder = find.byKey(Key('delete_icon_recipient_tag_item_${prefix.name}_0'));
>       final avatarIconRecipientTagItemWidgetFinder = find.byKey(Key('avatar_icon_recipient_tag_item_${prefix.name}_0'));
> 
>       final Size recipientTagItemWidgetSize = tester.getSize(recipientTagItemWidgetFinder);
>       final Size labelRecipientTagItemWidgetSize = tester.getSize(labelRecipientTagItemWidgetFinder);
>       final Size deleteIconRecipientTagItemWidgetSize = tester.getSize(deleteIconRecipientTagItemWidgetFinder);
>       final Size avatarIconRecipientTagItemWidgetSize = tester.getSize(avatarIconRecipientTagItemWidgetFinder);
> 
>       log('recipient_composer_widget_test::main: TagSize = $recipientTagItemWidgetSize | LabelTagSize = $labelRecipientTagItemWidgetSize | DeleteIconTagSize = $deleteIconRecipientTagItemWidgetSize | AvatarIconTagSize = $avatarIconRecipientTagItemWidgetSize');
> 
>       expect(labelRecipientTagItemWidgetFinder, findsOneWidget);
>       expect(deleteIconRecipientTagItemWidgetFinder, findsOneWidget);
>       expect(avatarIconRecipientTagItemWidgetFinder, findsOneWidget);
> 
>       expect(
>           labelRecipientTagItemWidgetSize.width + deleteIconRecipientTagItemWidgetSize.width + avatarIconRecipientTagItemWidgetSize.width,
>           lessThanOrEqualTo(recipientTagItemWidgetSize.width)
>       );
>     });
> ```

As far as I can see, you wrote the unit test with incorrect logic. If your `EmailAddress` has `displayName == null` then `AvatarIcon` will not have it thus leading to error `_KeyWidgetFinder:<Found 0 widgets with key [<'avatar_icon_recipient_tag_item_to_0'>]: `.
You can refer to the cases of email addresses being too long that I have written in the commit below.